include/deal.II/lac/vector.templates.h

```diff
@@ -170,13 +170,14 @@ Vector<Number>::Vector (const TrilinosWrappers::MPI::Vector &v)
       allocate();
 
       // Copy the distributed vector to
-      // a local one at all
-      // processors. TODO: There could
+      // a local one at all processors
+      // that know about the original vector.
+      // TODO: There could
       // be a better solution than
       // this, but it has not yet been
       // found.
       TrilinosWrappers::MPI::Vector localized_vector;
-      localized_vector.reinit(complete_index_set(vec_size));
+      localized_vector.reinit(complete_index_set(vec_size), v.get_mpi_communicator());
```

Comment:
Yes, this makes sense to me. What communicator did this use before? A communicator that has only a single participating process?

Reply:
`MPI_COMM_WORLD`:
```c++
void Vector< Number >::reinit ( 
  IndexSet & parallel_partitioning,
  const MPI_Comm & communicator = MPI_COMM_WORLD,
  const bool omit_zeroing_entries = false
  )
```

We offer the default choice of the global communicator in quite a number of places - I'm relieved that this is the first (and hopefully only?) place that I've fun into this issue. Well done to those that implemented our MPI-related interfaces for being so precise about propagating the communicators throughout the code :-)

Reply:
I wished we hadn't used defaults anywhere :-( It's just so easy to make the kind of mistake you just found out about...